### PR TITLE
MAINT: update scipy.linalg section

### PR DESCRIPTION
**Pull request by mdhaber**

### CODE REVIEW

File: intro/scipy/index.rst [hunk 1]
```diff
@@ -157,73 +157,72 @@ functions here. Frequently used ones are:
 Linear algebra operations: :mod:`scipy.linalg`
 ----------------------------------------------
 
-.. tip::
```
Comment (mdhaber) [inline intro/scipy/index.rst:160]: "Tip" is all over the place. Do you agree that this (and many of them) are unnecessary?

File: intro/scipy/index.rst [hunk 2]
```diff
@@ -157,73 +157,72 @@ functions here. Frequently used ones are:
...
-
-    The :mod:`scipy.linalg` module provides standard linear algebra
-    operations, relying on an underlying efficient implementation (BLAS,
-    LAPACK).
+:mod:`scipy.linalg` provides a Python interface to efficient, compiled
+implementations of standard linear algebra operations: the BLAS (Basic
+Linear Algebra Subroutines) and LAPACK (Linear Algebra PACKage) libraries.
```
Comment (mdhaber) [inline intro/scipy/index.rst:162]: This sentence could use additional wordsmithing.

File: intro/scipy/index.rst [hunk 3]
```diff
@@ -157,73 +157,72 @@ functions here. Frequently used ones are:
...
 
     >>> import scipy as sp
     >>> arr = np.array([[1, 2],
     ...                 [3, 4]])
     >>> sp.linalg.det(arr)
     -2.0
-    >>> arr = np.array([[3, 2],
-    ...                 [6, 4]])
+    >>> arr = np.array([[1, 1],
+    ...                 [1, 1]])
```
Comment (mdhaber) [inline intro/scipy/index.rst:173]: (even) more obviously singular

File: intro/scipy/index.rst [hunk 4]
```diff
@@ -157,73 +157,72 @@ functions here. Frequently used ones are:
...
-    ...                 [3, 4]])
-    >>> iarr = sp.linalg.inv(arr)
-    >>> iarr
-    array([[-2. ,  1. ],
-           [ 1.5, -0.5]])
-    >>> np.allclose(arr @ iarr, np.eye(2))
+    >>> A = np.array([[1, 2],
+    ...               [2, 3]])
+    >>> b = np.array([14, 23])
+    >>> x = sp.linalg.solve(A, b)
```
Comment (mdhaber) [inline intro/scipy/index.rst:187]: Using neater numbers and simpler variable names.

File: intro/scipy/index.rst [hunk 5]
```diff
@@ -157,73 +157,72 @@ functions here. Frequently used ones are:
...
-    >>> arr = np.array([[3, 2],
-    ...                 [6, 4]])
+    >>> arr = np.array([[1, 1],
+    ...                 [1, 1]])
     >>> sp.linalg.det(arr)
     0.0
     >>> sp.linalg.det(np.ones((3, 4)))
     Traceback (most recent call last):
     ...
     ValueError: expected square matrix
```
Comment (mdhaber) [inline intro/scipy/index.rst:179]: Is this important to emphasize?

File: intro/scipy/index.rst [hunk 6]
```diff
@@ -157,73 +157,72 @@ functions here. Frequently used ones are:
...
     Traceback (most recent call last):
     ...
-    ...LinAlgError: singular matrix
-
-* More advanced operations are available, for example singular-value
-  decomposition (SVD)::
-
-    >>> arr = np.arange(9).reshape((3, 3)) + np.diag([1, 0, 1])
-    >>> uarr, spec, vharr = sp.linalg.svd(arr)
+    ...LinAlgError: Matrix is singular
```
Comment (mdhaber) [inline intro/scipy/index.rst:201]: Is this important to emphasize? (Should we remove it?)

File: intro/scipy/index.rst [hunk 7]
```diff
@@ -157,73 +157,72 @@ functions here. Frequently used ones are:
...
     0.0
     >>> sp.linalg.det(np.ones((3, 4)))
     Traceback (most recent call last):
     ...
     ValueError: expected square matrix
 
-* The :func:`scipy.linalg.inv` function computes the inverse of a square
-  matrix::
+The :func:`scipy.linalg.solve` function solves linear systems of the
+form :math:`Ax = b`.
```
Comment (mdhaber) [inline intro/scipy/index.rst:182]:
```suggestion
Mathematically, the solution of a linear system :math:`Ax = b` is :math:`x = A^{-1}b`,
but explicit inversion of a matrix is numerically unstable and should be avoided.
Instead, use :func:`scipy.linalg.solve`::
```

File: intro/scipy/index.rst [hunk 8]
```diff
@@ -157,73 +157,72 @@ functions here. Frequently used ones are:
...
+    >>> A = np.array([[1, 2],
+    ...               [2, 3]])
+    >>> b = np.array([14, 23])
+    >>> x = sp.linalg.solve(A, b)
+    >>> x
+    array([4., 5.])
+    >>> np.allclose(A @ x, b)
     True
 
-  Finally computing the inverse of a singular matrix (its determinant is zero)
```
Comment (mdhaber) [inline intro/scipy/index.rst:195]: Is there a need to indent text within a section?

File: intro/scipy/index.rst [hunk 9]
```diff
@@ -157,73 +157,72 @@ functions here. Frequently used ones are:
...
 
-    >>> sarr = np.diag(spec)
-    >>> svd_mat = uarr @ sarr @ vharr
-    >>> np.allclose(svd_mat, arr)
+    >>> S = np.diag(s)  # convert to diagonal matrix before matrix multiplication
+    >>> A2 = U @ S @ Vh
+    >>> np.allclose(A2, A)
     True
+    >>> A3 = (U * s) @ Vh  # more efficient: use array math broadcasting rules!
+    np.allclose(A3, A)
```
Comment (mdhaber) [inline intro/scipy/index.rst:220]:
```suggestion
    >>> np.allclose(A3, A)
    True
```

**Comment by jarrodmillman** [comment]
I am going to merge this and you can remove the example showing that it fails on non-square matrices in a follow-up PR.

**Pull request merged**